dist/templates/k8s.ovn.org_egressips.yaml.j2

```diff
@@ -0,0 +1,139 @@
+
+---
```

Comment:
you can just make OVN register the CRD if the config option is enabled. Take a look at this example:

https://github.com/nimbess/stargazer/blob/master/cmd/stargazer/main.go#L79

Reply:
I actually didn't understand that example. 

`CreateCRD` does not create your full CRD object, it only seems to create a subset of it with `Spec.Group`, `Spec.Versions`, `Spec.Scope` and `Spec.Names` being set: https://github.com/nimbess/stargazer/blob/master/pkg/crd/api/unp/v1/types.go#L66  

Moreover, the egress IP feature is disabled by default, I thus think it's more appropriate to not create the CRD at all and only have the user create it when he/she enables the feature him/her-self. The full CRD is distributed anyways in `dist/templates/k8s.ovn.org_egressips.yaml.j2`, so it would be a matter of running one command.

Reply:
As we discussed, you would just need to add the Validation portion to the CRD object, although that's a real pain to do by hand. Not sure if there is a way to get it from the custom type or some library to create it for you for the open api schema. Otherwise, you could just assign your CRD yaml output to a var and embed it in a go file, then issue the client create call with the yaml. Either option imo is better than relying on CNO or KIND to register the CRD for you.

Reply:
Due to the offline remarks about the security concern for the ovn-k8s pod doing the CRD registering I'm OK with it living outside of ovn-k8s.